bindings_node/src/mls_client.rs

```diff
@@ -0,0 +1,189 @@
+
+  let xmtp_client = ClientBuilder::new(identity_strategy)
+    .api_client(api_client)
+    .store(store)
+    .build()
+    .await
+    .map_err(|e| Error::from_reason(format!("{}", e)))?;
+
+  Ok(NapiClient {
+    inner_client: Arc::new(xmtp_client),
+    account_address,
+  })
+}
+
+#[napi]
+impl NapiClient {
+  #[napi]
+  pub fn inbox_id(&self) -> String {
+    self.inner_client.inbox_id()
+  }
+
+  #[napi]
+  pub fn is_registered(&self) -> bool {
+    self.inner_client.identity().signature_request().is_none()
+  }
+
+  #[napi]
+  pub fn installation_id(&self) -> String {
+    ed25519_public_key_to_address(self.inner_client.installation_public_key().as_slice())
+  }
+
+  #[napi]
+  pub async fn can_message(&self, account_addresses: Vec<String>) -> Result<HashMap<String, bool>> {
+    let results: HashMap<String, bool> = self
+      .inner_client
+      .can_message(account_addresses)
+      .await
+      .map_err(|e| Error::from_reason(format!("{}", e)))?;
+
+    Ok(results)
+  }
+
+  #[napi]
+  pub async fn register_ecdsa_identity(&self, signature_bytes: Uint8Array) -> Result<()> {
+    if self.is_registered() {
+      return Err(Error::from_reason(
+        "An identity is already registered with this client",
+      ));
+    }
+
+    let mut signature_request = match self.inner_client.identity().signature_request() {
+      Some(signature_req) => signature_req,
+      None => return Err(Error::from_reason("No signature request found")),
+    };
+    let signature_text = match self.signature_text() {
+      Some(text) => text,
+      None => return Err(Error::from_reason("No signature text found")),
+    };
+
+    let signature = Box::new(RecoverableEcdsaSignature::new(
+      signature_text,
+      signature_bytes.deref().to_vec(),
+    ));
+
+    signature_request
+      .add_signature(signature)
+      .await
+      .map_err(|e| Error::from_reason(format!("{}", e)))?;
+
+    self
+      .inner_client
+      .register_identity(signature_request)
+      .await
+      .map_err(|e| Error::from_reason(format!("{}", e)))?;
+
+    return Ok(());
+  }
+
+  #[napi]
+  pub async fn register_erc1271_identity(
+    &self,
+    signature_bytes: Uint8Array,
+    chain_rpc_url: String,
+  ) -> Result<()> {
+    if self.is_registered() {
+      return Err(Error::from_reason(
+        "An identity is already registered with this client",
+      ));
+    }
+
+    let mut signature_request = match self.inner_client.identity().signature_request() {
+      Some(signature_req) => signature_req,
+      None => return Err(Error::from_reason("No signature request found")),
+    };
+    let signature_text = match self.signature_text() {
+      Some(text) => text,
+      None => return Err(Error::from_reason("No signature text found")),
+    };
+
+    let signature = Box::new(
+      Erc1271Signature::new_with_rpc(
+        signature_text,
+        signature_bytes.deref().to_vec(),
+        self.account_address.clone(),
```

Comment:
we should take `account_address` from arguments instead of client. Client should represent inbox instead of an account address.